docs/extending_deptrac.md

```diff
@@ -0,0 +1,96 @@
+# Extending Deptrac
+
+Deptrac defines its extension point by providing set of **contract** classes
+that you can depend on for your implementation. The classes can be found in
+the `src/Contract` directory and are covered by
+the [backwards compatibility policy](bc_policy.md).
+
+There are several ways you can extend Deptrac:
+
+- [Output formatters](#Output Formatters) - To change how the output of
+  dependency analysis is formatted. This might be useful to integrate Deptrac
+  into your CI/CD pipeline.
+- [Analyser Events](#Analyser Events) - Decide whether a dependency is
+  Uncovered, Allowed, Violation or SkippedViolation.
+- [Layer collectors](#Layer collectors) - Add custom deptrac collector
+
+> **Note**
+> In examples where FQCN is not specified, the base
+> namespace `Qossmic\Deptrac\Contract\` is omitted for readability.
+
+## Output Formatters
+
+Creating an output formatter requires only to create a new class implementing
+the `OutputFormatter\OutputFormatterInterface` and
```

Comment:
This might be hard to catch for people who don't know Deptrac's code well. I think this should be either `OutputFormatterInterface` to show, that it is not  in fact a FQCN or really use the FQCN instead.

Reply:
I mean, the note about this is literally 5 lines up.

Reply:
Yes, but I think it's reasonable for people to see `OutputFormatter\OutputFormatterInterface` and assume this is already a FQCN. I think writing either the FQCN or just reducing it to the class name will make it easier to process for people. On the other hand, if you are extending deptrac we can assume you know or look at the classes in more detail anyway. Let's keep it for now and if someone is  confused or bothered we can change it later.